tools/integration_tests/inactive_stream_timeout/setup_test.go

```diff
@@ -32,83 +32,73 @@ import (
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/test_suite"
 	"github.com/stretchr/testify/require"
 )
 
 const (
-	kTestDirName                         = "inactiveReadTimeoutTest"
+	kTestDirName                         = "inactiveReadTimeout"
 	kOnlyDirMounted                      = "onlyDirInactiveReadTimeout"
 	kFileSize                            = 10 * 1024 * 1024 // 10 MiB
-	kChunkSizeToRead                     = 128 * 1024       // 64 KiB
+	kChunkSizeToRead                     = 128 * 1024       // 128 KiB
 	kTestFileName                        = "foo"
 	kDefaultInactiveReadTimeoutInSeconds = 1 // A short timeout for testing
-	kLogFileNameForMountedDirectoryTests = "/tmp/inactive_stream_timeout_logs/log.json"
-	kHTTP1ClientProtocol                 = "http1"
-	kGRPCClientProtocol                  = "grpc"
+	GKETempDir                           = "/gcsfuse-tmp"
+	OldGKElogFilePath                    = "/tmp/inactive_stream_timeout_logs/log.json"
 )
 
-var (
-	// Stores test directory path in the mounted path.
-	gTestDirPath string
-
-	// Used to run the test for different types of mount by modify this function.
-	gMountFunc func([]string) error
-
-	// Actual mounted directory, for dynamic mount it becomes gRootDir/<bucket_name>
-	gMountDir string
-
-	// Root directory which is mounted by gcsfuse.
-	gRootDir string
+type env struct {
+	storageClient *storage.Client
+	ctx           context.Context
+	testDirPath   string
+	cfg           *test_suite.TestConfig
+	bucketType    string
+}
 
-	// Clients to create the object in GCS.
-	gStorageClient *storage.Client
-	gCtx           context.Context
+var (
+	testEnv   env
+	mountFunc func(*test_suite.TestConfig, []string) error
+	// mount directory is where our tests run.
+	mountDir string
+	// root directory is the directory to be unmounted.
+	rootDir string
 )
 
-type gcsfuseTestFlags struct {
-	cliFlags            []string
-	inactiveReadTimeout time.Duration
-	fileName            string
-	clientProtocol      string
+func SetupNestedTestDir(path string, permission os.FileMode, t *testing.T) {
+	err := os.MkdirAll(path, permission)
+	require.NoError(t, err)
 }
 
-func setupFile(ctx context.Context, storageClient *storage.Client, fileName string, fileSize int, t *testing.T) string {
-	t.Helper()
-	client.SetupFileInTestDirectory(ctx, storageClient, kTestDirName, fileName, int64(fileSize), t)
-	return path.Join(gTestDirPath, fileName)
-}
-
-func loadLogLines(reader io.Reader) ([]string, error) {
-	content, err := io.ReadAll(reader)
-	if err != nil {
-		return nil, err
+func mountGCSFuseAndSetupTestDir(flags []string, ctx context.Context, storageClient *storage.Client) {
+	setup.MountGCSFuseWithGivenMountWithConfigFunc(testEnv.cfg, flags, mountFunc)
+	// In case of GKE, the test directory is not created by the test.
+	if setup.MountedDirectory() != "" {
+		testEnv.testDirPath = path.Join(setup.MntDir(), kTestDirName)
```

Comment:
should it be this instead?
```suggestion
	if testEnv.cfg.GKEMountedDirectory != "" {
		setup.SetMntDir(testEnv.cfg.GKEMountedDirectory)
```

and then regardless of the mntDir path you always setup the test directory?